### ISSUE TITLE
Prover balance < commitment bond

### ISSUE THREAD

**Issue by jackhack00** [BlockScience/aztec-gddt#91]
Reworking the commit bond dynamics requires also selecting only an Agent (Prover) if their balance is above commitment bond size.
Currently, any Prover Agent can be chosen.

**Comment by jackhack00** [comment]
currently only have:
agents[prover_uuid].balance += rewards_prover
to increase balance. 
For this issue, we need to check whether current balance is high enough to put up commit bond >0

**Comment by jackhack00** [comment]
Needs an additional rule that we only pick from provers who have a `balance >= commit_bond_amount_from_params`
  @eightarmsninebrains can you fix this?
  
  Currently, we have below to pick Prover. 
  ```
if bernoulli_trial(params['proving_marketplace_usage_probability']) is True:
      provers: list[AgentUUID] = [
                                  a_id 
                                  for (a_id, a) 
                                  in state['agents'].items() 
                                  if a.is_prover]
      # XXX: relays are going to be uniformly sampled
      prover: AgentUUID = choice(provers)
      bond_amount = commit_bond_amount_from_params
```

**Comment by jackhack00** [comment]
Closed with #104
